File: frontend/src/api/ai.ts
import { apiDelete, apiGet, apiPatch, apiPost } from './client';

export interface AiApiConfig {
  id: number;
  name: string;
  api_type: 'anthropic' | 'openai';
  url: string | null;
  model: string;
  effort: 'low' | 'medium' | 'high' | null;
  request_timeout: number;
  batch_size: number | null;
  reasoning_enabled: boolean;
  key_set: boolean;
  key_hint: string | null;
  created_at: string;
}

export interface AiApiConfigCreate {
  name: string;
  api_type: 'anthropic' | 'openai';
  url?: string | null;
  model?: string;
  effort?: 'low' | 'medium' | 'high' | null;
  request_timeout?: number;
  batch_size?: number | null;
  reasoning_enabled?: boolean;
  api_key?: string | null;
}

export type AiApiConfigUpdate = Partial<AiApiConfigCreate>;

const BASE = '/api/settings/ai-apis';

export function listAiApiConfigs(): Promise<AiApiConfig[]> {
  return apiGet(BASE);
}

export function createAiApiConfig(body: AiApiConfigCreate): Promise<AiApiConfig> {
  return apiPost(BASE, body);
}

export function updateAiApiConfig(id: number, body: AiApiConfigUpdate): Promise<AiApiConfig> {
  return apiPatch(`${BASE}/${id}`, body);
}

export function deleteAiApiConfig(id: number): Promise<void> {
  return apiDelete(`${BASE}/${id}`);
}

export interface AiTextGenerateRequest {
  entity_type: string;
  existing_fields?: Record<string, string>;
  prompt: string;
}

export interface AiTextGenerateResponse {
  draft: string;
}

export function generateAiText(body: AiTextGenerateRequest): Promise<AiTextGenerateResponse> {
  return apiPost('/api/ai/generate', body);
}

File: backend/app/schemas/ai.py
"""AiApiConfig schemas (DHCM-95): named AI API endpoint configs for AI text
generation. Modeled directly on STL Studio's AiApiConfig schemas.

Encrypted keys are never returned -- only whether one is set and a masked hint.
"""

from datetime import datetime

from pydantic import BaseModel, Field


class AiApiConfigCreate(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    api_type: str = Field(pattern="^(anthropic|openai)$")
    url: str | None = Field(default=None, max_length=500)
    model: str = Field(default="", max_length=200)
    effort: str | None = Field(default=None, pattern="^(low|medium|high)$")
    request_timeout: int = Field(default=10, ge=1, le=600)
    batch_size: int | None = Field(default=None, ge=1, le=50)
    reasoning_enabled: bool = False
    # Optional so a config can still be created key-less (e.g. a local model
    # with no auth), but lets the client set the key in the same request
    # instead of a follow-up call.
    api_key: str | None = Field(default=None, max_length=400)


class AiApiConfigUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=100)
    url: str | None = Field(default=None, max_length=500)
    model: str | None = Field(default=None, max_length=200)
    effort: str | None = Field(default=None, pattern="^(low|medium|high)$")
    request_timeout: int | None = Field(default=None, ge=1, le=600)
    batch_size: int | None = Field(default=None, ge=1, le=50)
    reasoning_enabled: bool | None = None
    api_key: str | None = Field(default=None, max_length=400)

    model_config = {"extra": "forbid"}


class AiApiConfigOut(BaseModel):
    id: int
    name: str
    api_type: str
    url: str | None
    model: str
    effort: str | None
    request_timeout: int
    batch_size: int | None
    reasoning_enabled: bool
    key_set: bool
    key_hint: str | None
    created_at: datetime
